ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml

```diff
@@ -28,7 +28,40 @@ DappsComboBox {
     }
 
     onDisconnectDapp: (dappUrl) => {
-        root.wcService.disconnectDapp(dappUrl)
+        disconnectdAppDialogLoader.dAppUrl = dappUrl
+        disconnectdAppDialogLoader.active = true
+    }
+
+    Loader {
```

Comment:
Why not put your popup in Popups.qml and use the standard `Global.openPopup(...)` instead of this manual `Loader`?

Reply:
Question still remains :) Why the `Loader` here?

Reply:
Looks like it's following the approach used in the DAppsWorkflow. I don't think it's wrong, it's just an intermediary state in the dev process.
We've had a discussion on this some time ago and agreed to leave the DAppsWorkflow self-contained during the initial development. I think it helps in better controlling the flow and dependencies and to also speed-up the development. If you think about it all the `DAppsWorkflow` needs to be broken into pieces. Right now it's a ComboBox that holds the entire WalletConnect feature. Once we'll have a stable and somewhat completed feature we can start moving the building blocks.